src/thermo/CoverageDependentSurfPhase.cpp

```diff
@@ -0,0 +1,772 @@
+}
+
+// Functions calculating reference state thermodyanmic properties--------------
+
+void CoverageDependentSurfPhase::getGibbs_RT_ref(doublereal* grt) const
+{
+    _updateReferenceThermo();
+    scale(m_mu_ref.begin(), m_mu_ref.end(), grt, 1.0/RT());
+}
+
+void CoverageDependentSurfPhase::getEnthalpy_RT_ref(doublereal* hrt) const
+{
+    _updateReferenceThermo();
+    scale(m_h_ref.begin(), m_h_ref.end(), hrt, 1.0/RT());
+}
+
+void CoverageDependentSurfPhase::getEntropy_R_ref(doublereal* sr) const
+{
+    _updateReferenceThermo();
+    scale(m_s_ref.begin(), m_s_ref.end(), sr, 1.0/GasConstant);
+}
+
+void CoverageDependentSurfPhase::getCp_R_ref(doublereal* cpr) const
+{
+    _updateReferenceThermo();
+    scale(m_cp_ref.begin(), m_cp_ref.end(), cpr, 1.0/GasConstant);
+}
+
+// Functions calculating standard state thermodyanmic properties---------------
+
+void CoverageDependentSurfPhase::getEnthalpy_RT(doublereal* hrt) const
+{
+    _updateThermo();
+    scale(m_enthalpy.begin(), m_enthalpy.end(), hrt, 1.0/RT());
+}
+
+void CoverageDependentSurfPhase::getEntropy_R(doublereal* sr) const
+{
+    _updateThermo();
+    scale(m_entropy.begin(), m_entropy.end(), sr, 1.0/GasConstant);
+    if (m_has_ref_coverage) {
+        double tmp = -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            sr[k] -= tmp;
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getCp_R(doublereal* cpr) const
+{
+    _updateThermo();
+    scale(m_heatcapacity.begin(), m_heatcapacity.end(), cpr, 1.0/GasConstant);
+}
+
+void CoverageDependentSurfPhase::getGibbs_RT(doublereal* grt) const
+{
+    _updateThermo();
+    scale(m_chempot.begin(), m_chempot.end(), grt, 1.0/RT());
+    if (m_has_ref_coverage) {
+        double tmp = -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            grt[k] += tmp;
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getPureGibbs(doublereal* g) const
+{
+    getGibbs_RT(g);
+    for (size_t k = 0; k < m_kk; k++) {
+        g[k] *= RT();
+    }
+}
+
+void CoverageDependentSurfPhase::getStandardChemPotentials(doublereal* mu0) const
+{
+    _updateThermo();
+    copy(m_chempot.begin(), m_chempot.end(), mu0);
+    if (m_has_ref_coverage) {
+        double tmp = RT() * -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            mu0[k] += tmp;
+        }
+    }
+}
+
+// Functions calling partial molar thermodyanmic properties----------------
+
+void CoverageDependentSurfPhase::getPartialMolarEnthalpies(doublereal* hbar) const
+{
+    _updateThermo();
+    copy(m_enthalpy.begin(), m_enthalpy.end(), hbar);
+}
+
+void CoverageDependentSurfPhase::getPartialMolarEntropies(doublereal* sbar) const
+{
+    _updateThermo();
+    copy(m_entropy.begin(), m_entropy.end(), sbar);
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            sbar[k] -= GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref);
+        }
```

Comment:
If you end up introducing an `m_cov` array, accessing that could replace these calls to `concentration` and `size`, both of which incur some unnecessary overhead.